### PR TITLE
fixed the code according to the reviews

### PR DESCRIPTION
**Pull request by baram104**

### CODE REVIEW

File: Components/Form.tsx [hunk 1]
```diff
@@ -10,29 +10,33 @@ import {
...
+  const validateInput = (value: string | undefined) => {
+    setIsInputTouched(true);
+    if (value) {
+      setIsFormValid(true);
+      return true;
+    } else {
       setIsFormValid(false);
-      return;
+      return false;
     }
```
Comment (yehorGuyAvrahamCom) [inline Components/Form.tsx:29]: If it's a `required` validation I recommend using the next syntax 
`
setIsFormValid(Boolean(value))
`
And you don't need to return any value from this function and just use the state value in the submit function ;)

Comment (baram104) [reply Components/Form.tsx:29]: That's a cool syntax indeed , thanks!

**Comment by EkkoKo** [comment]
I started going through your code, and I've got a couple of questions.
Why would you make the registration steps as a dynamic route?
When you only have 2 inputs - `full name` and `date of birth` , isn't a better way to manage those pages is to have two separate pages `/fullname` and `/birthdate`?
After going through it more I tried to only set my localStorage to have the date of birth object which made the webside redirect me to a NaN route.
image
Wouldn't it be better  to validate the objects that are stored in localStorage to check if the steps are valid and not play with array length?

**Comment by baram104** [comment]
Well, about the first question, i was considering this option, though eventually i decided to do it as dynamic as possible in case you would need to add more registration steps. Regarding the second question, it actually might be better to store them separately in case the user sets the local storage manually. 
However, validation based on local storage isn’t a bad practice anyway?

**Comment by EkkoKo** [comment]
Even if we would of added more routes don't you think it's a better practice to have everything declarative and very easy to understand more than just numbering steps?
About modifying the localStorage that's what I did and that's how I managed to find the bug that's being caused because you're validating information based on your array length.
And in some cases localStorage validation could be a bad practice but I think that in a situation where you only depend of localStorage and your state management you should validate that the data is exactly what you expect it to be and that way bulletproof yourself against small changes like modifying the localStorage.

**Comment by baram104** [comment]
Yea it makes sense! Thanks!

**Pull request merged**